### ISSUE TITLE
It doesn't go to definition of a function with an negative optional parameter

### ISSUE THREAD

**Issue by megaslav** [cweijan/autohotkey-plus#5]
Thanks for this great project, I use it with pleasure!

I have the following function: 

```
RandomSleep(timez:=300, rand:=-1) { 
  ; some code here
}
```

If I call this function anywhere in the code and try to go to its definition with F12 or Ctrl + Click, it doesn't work. It say that definition is not found.

If I remove the `-` character from `rand` it works. This works:

`RandomSleep(timez:=300, rand:=1)`

Very strange.

And for this one I cannot get Go To Definition work even if I remove both - from the definition:

```
DeltaTickS(FirstTick:= -1, SecondTick:= -1){
    if (FirstTick = -1)
        FirstTick:= A_TickCount
    if (SecondTick = -1)
        SecondTick:= A_TickCount
    return ((FirstTick - SecondTick) // 1000)
}
```

Creating an empty `DeltaTicksS() { }` works though. Even highlighting the definition of my original function and pressing F12 takes to this new empty function.

**Comment by CoffeeChaton** [comment]
https://github.com/cweijan/vscode-autohotkey/blob/master/src/core/Detecter.ts#L13
```
 //   private static methodPattern = /(([\w_]+)\s*\([\w\s,:"=]*\))\s*\{/;
//    private static methodSecondPattern = /(([\w_]+)\s*\([\w\s,:"=]*\))\s*/;
private static methodPattern =  /(([\w_]+)\s*\([\w\s,:"=+\-*/\\`]*\))\s*\{/;
private static methodSecondPattern = /(([\w_]+)\s*\([\w\s,:"=+\-*/\\`]*\))\s*/;
```

this match can't search like 
```
foo(a:='}',b :='{',c := ')', d := '(', e:= '*',f = 'a+b',g := "/"){
;nothing
}
```

and also can't find
```
MDMF_FromPoint(ByRef X := "", ByRef Y := "", Flag := 0) 
{    If (X)
     ; some code here
     ; some code here
}
```

because of  `{ if`

**Comment by CoffeeChaton** [comment]
i fix this bug at my fork version.
at this weekend, i will change my code style like cweijan style and push request.

**Comment by cweijan** [comment]
@CoffeeChaton  Thanks for notice , fix it at version 2.3.0

**Comment by megaslav** [comment]
Thanks for the quick fix!!
